api/app/src/routes/user.ts

```diff
@@ -112,6 +113,32 @@ router.get(
   })
 );
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function revokeUserProviderAccess(connectedUser: ConnectedUser, provider: any) {
+  const providerOAuth2 = providerOAuth2OptionsSchema.safeParse(provider);
+
+  // TODO #249: implement garmin revoke support
+  // const providerOAuth1 = providerOAuth1OptionsSchema.safeParse(
+  //   req.query.provider
+  // );
+
+  if (providerOAuth2.success) {
+    await Constants.PROVIDER_OAUTH2_MAP[providerOAuth2.data].revokeProviderAccess(connectedUser);
+
+    return true;
+    // } else if (providerOAuth1.success) {
+    //   // await Constants.PROVIDER_OAUTH1_MAP[
+    //   //   providerOAuth1.data
+    //   // ].deregister(connectedUser);
+  } else if (provider === PROVIDER_APPLE) {
+    const apple = new Apple();
+    await apple.revokeProviderAccess(connectedUser);
+    return true;
+  } else {
+    throw new BadRequestError(`Provider not supported: ${provider}`);
+  }
```

Comment:
NABD, but could remove this last `else` and have the throw as the default, catch-all option.